mitiq/factories.py

```diff
@@ -319,4 +329,89 @@ def reduce(self) -> float:
 
         return self.static_reduce(
             self.instack, self.outstack, self.asymptote, self.order,
+        )[0]
+
+
+class AdaExpFactory(Factory):
+    """Factory object implementing an adaptive zero-noise extrapolation algorithm assuming an
+    exponential ansatz y(x) = a + b * exp(-c * x), with c > 0.
+
+    The noise scale factors are are chosen adaptively at each step, depending on the history
+    of collected results.
+
+    If the asymptotic value (y(x->inf) = a) is known, a linear fit with respect
+    to z(x) := log[sing(b) (y(x) - a)] is used.
+    Otherwise, a non-linear fit of y(x) is performed.
+    """
+
+    SHIFT_FACTOR = 1.27846
```

Comment:
```suggestion
    _SHIFT_FACTOR = 1.27846
```

Reply:
+ later changes. If this is constant it shouldn't be visible outside the class.

Reply:
Ok, changed.